tests/core/assertions.test.js

```diff
@@ -197,3 +206,64 @@ test('check object fully matches spec', () => {
         'Expected json response to fully match spec, but it does not'
     )
 })
+
+test('check object property equals dateOffset and format', () => {
+    const object = {
+        beginDate: '2018-05-04'
+    }
+    const spec = [
+        {
+            field: 'beginDate',
+            matcher: 'dateOffset(1days,fr)',
+            value: 'YYYY-MM-DD'
+        },
+        {
+            field: 'beginDate',
+            matcher: 'dateOffset(+2days,fr)',
+            value: 'YYYY-MM-DD'
+        },
+        {
+            field: 'beginDate',
+            matcher: 'dateOffset(-2days,fr)',
+            value: 'YYYY-MM-DD'
+        },
+        {
+            field: 'beginDate',
+            matcher: 'dateOffset(-2days,fr)',
+            value: "[Aujourd'hui] YYYY-MM-DD hh[h]mm"
+        }
+    ]
+
+    assertObjectMatchSpec(object, spec)
+    expect(require('chai').expect).toHaveBeenCalledWith(
+        '2018-05-04',
+        `Expected property 'beginDate' to equal '2018-05-02', but found '2018-05-04'`
+    )
+    expect(require('chai').expect).toHaveBeenCalledWith(
+        '2018-05-04',
+        `Expected property 'beginDate' to equal '2018-05-03', but found '2018-05-04'`
+    )
+    expect(require('chai').expect).toHaveBeenCalledWith(
+        '2018-05-04',
+        `Expected property 'beginDate' to equal '2018-04-29', but found '2018-05-04'`
+    )
+    expect(require('chai').expect).toHaveBeenCalledWith(
+        '2018-05-04',
+        `Expected property 'beginDate' to equal 'Aujourd'hui 2018-04-29 12h00', but found '2018-05-04'`
+    )
+})
+
+test('check dateOffset throw Exception given invalid locale', () => {
```

Comment:
You could add a test for invalid unit too, but maybe it's in progress

Reply:
Yes, I was in progress and found that we can't really valid it except thtough the regex ( which validates it only partially). When applying amount and unit with the `add` function of moment, the date stills stay the same and won't be invalid though unit and amount are ambiguous or invalid. 

To me, for the amount, the regex validates it as it must be number. Regarding the unit, we can create a whitelist. This list will be this one: https://momentjs.com/docs/#/manipulating/add/

Reply:
Well… we'll have to add singular plural, shortand, maybe it's fine as it is :)

Reply:
Agreed, we can maybe be stricter regarding the regex for the unit as it must be at least one character and a maximum of 12?

Reply:
not sure it helps as this value is not very human friendly, you'll have no real clue on what's wrong.